src/main/java/tech/jhipster/lite/module/domain/JHipsterModulesApplyer.java

```diff
@@ -109,6 +110,14 @@ public JHipsterModuleApplied apply(JHipsterModuleToApply moduleToApply) {
     return moduleApplied;
   }
 
+  private JHipsterModulePackageJson buildPackageJson(JHipsterModule module, Optional<JavaBuildTool> detectedJavaBuildTool) {
+    JHipsterModuleContext context = detectedJavaBuildTool
```

Comment:
I think this would be clearer if we extract to a method:
```java
 private Optional<JavaBuildTool> detectedJavaBuildTool(JHipsterModule module) {
    return javaBuildTools
      .detect(module.projectFolder())
      .or(() -> javaBuildTools.detect(module.files()));
}
```

And then use it here and in `buildStartupCommands(..)` (by changing its signature to `buildStartupCommands(JHipsterModule)`), and remove it from the `Optional<JavaBuildTool>` as an input parameter:
```suggestion
    JHipsterModuleContext context = detectedJavaBuildTool(module)
```